addons/web/static/src/legacy/js/core/utils.js

```diff
@@ -240,9 +240,58 @@ class UnknownPatchError extends Error {
     }
 }
 
+const __escape = _.escape;
+_.escapeMethod = Symbol('html')
+_.escape = function escape(s) {
+    return s[_.escapeMethod] ? s[_.escapeMethod]() : __escape(s);
+}
+
+// notable issues:
+// * objects can't be negative in JS, so !!"" -> false but
+//   !!(new String) -> true, likewise markup
+// TODO (?)
+// * Markup.join / Markup#join => escapes items and returns a Markup
+// * Markup#replace => automatically escapes the replacements (difficult impl)
+class _Markup extends String {
+    [_.escapeMethod]() {
+        return this;
+    }
+}
+// exposed for qweb2.js
+window._Markup = _Markup;
+
+/**
+ * Returns a markup object, which acts like a String but is considered safe by
+ * `_.escape` (and will thus not be re-escaped).
+ *
+ * Can be used as a template function, or as a regular wrapper.
+ *
+ * Note: if the markup value is empty, will return an empty string instead of a
+ *       Markup, to avoid truthy empty Markups
+ *
+ * Note 2: unlike `markupsafe.Markup`, `Markup` doesn't generally override
+ *         string operations as most of them can't be overridden
+ */
+function Markup(v, ...exprs) {
```

Comment:
could you add docstrings? I'm really not sure to understand how this works

Reply:
There's a docstring? I don't think the types will help any given what they are.

This is implemented to work as a regular wrapper e.g. `Markup(content)` in which case `v` is a string, or as a template tag e.g.

    Markup`foo ${bar}`

in which case `v` is an `Array` of literal segments, and the substitutions are passed in as additional parameters.

Reply:
ok this makes sense. Maybe add this to the docstring then, i.e. elaborate a bit `Can be used as a template function, or as a regular wrapper.` with examples

Reply:
Fair enough, I'll do that.